### DISCUSSION TITLE
Not sure whose mistake this is. Filter dont working

### DISCUSSION THREAD

**Discussion by Gavolot** [bevyengine/bevy#11086] [Q&A] [answered]
## Bevy version
0.12.1

I'm sure that the input works. I checked if it works without entity iteration.

I tried to do a lot. But I don’t understand why the filter simply doesn’t seem to find my essence... It’s like it doesn’t exist. I don't understand.

I cant move player...
I just want the player to move 150 units somewhere =)
But I do everything as in the textbooks, but the component seems to be missing.  However, the "sprite" will spawn.

I've already gone crazy searching.

![image](https://github.com/bevyengine/bevy/assets/3579229/546b259d-13df-4dc7-9958-9385d113fd18)
![image](https://github.com/bevyengine/bevy/assets/3579229/73303a25-b825-482c-9f1a-3dfe52a92ef5)
![image](https://github.com/bevyengine/bevy/assets/3579229/3c3d068f-cdf5-4367-a5c5-9072f4ce2e7f)
![image](https://github.com/bevyengine/bevy/assets/3579229/99ec0de6-4508-4037-830e-76ceb74f7ac7)
![image](https://github.com/bevyengine/bevy/assets/3579229/a1a006dd-06bf-4422-959d-cc66015e7b6c)

**Comment by pablo-lua** [comment]
Can you maybe query the Player too with &Player, just so we can know that the problem is indeed the Filter?
Maybe something like:
```rust
query_player: Query<&Player>
// the function here...
for player in query_player {
   println!("{}", player.input_id);
}
```

**Reply by Gavolot** [reply]
![image](https://github.com/bevyengine/bevy/assets/3579229/692b6b80-3386-4218-aec3-1f4abc483e1b)
Unfortunately it didn't work.
Now I move on to check the answer below, which says that I may be "duplicating a type or content".

But I don't understand how I can "duplicate" the type itself if it checks "by type".

**Reply by pablo-lua** [reply]
The problem is that you are using The Player from two different mods, and the compiler is saying, "Hey, that's a different type."
You must assert that you are using the Player from the same mod (at least that is what I got from the answer below)

**Comment by akimakinai** [comment]
You seem to be dealing with many different `Player` types, as multiple `path` attributes to the same file duplicates the content.

Example:

```rust
#[path="player.rs"]
mod a;

#[path="player.rs"]
mod b;

fn main() {
    let a = a::Player;
    a == b::Player; // error: mismatched types
    // ^note: `b::Player` and `a::Player` have similar names, but are actually distinct types
}
```

`#[path]` is a niche feature to manually tackle special file structures. See [the book](https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html) or [the by-example](https://doc.rust-lang.org/rust-by-example/mod/split.html) for how you usually can split files in Rust, if the above is not what you wanted.

**Reply by akimakinai** [reply]
If that was not clear, the code above is simply expanded to this (IIRC):

```rust
mod a {
    // content of `player.rs`
    pub struct Player;
}

mod b {
    // content of `player.rs`
   pub struct Player;
}
```

What compiler actually sees is you declared two different types, which coincidentally have the same name.

You need only a single `mod` for the `player` module. If you have `mod player;` in your crate root, then you can refer to it from anywhere using `crate::player::`.

**Reply by Gavolot** [reply]
I still don’t understand why it’s different if I literally wrote down the path to these files.

Those. I understood what happened, but I still don’t understand how to “bypass duplication”. And just add the component.

**Reply by Gavolot** [reply]
![image](https://github.com/bevyengine/bevy/assets/3579229/ec420b4c-2473-44d4-9d86-c04a2d632ea2)
I'm trying, but none of this works, including the generation of cargo boxes))

How was it possible to make such a complex import from folders? I've never seen anything like this before.

**Reply by pablo-lua** [reply, marked as answer]
You should go back to where you were before, in the first prints, and maybe just try to use the same mod. Maybe defining the Player mod upper in the main file, then doing `use crate::player::Player;` (something like this)

**Reply by Gavolot** [reply]
I have now “calmed down” and sat in another empty project. I think I realized a few things. But I will note when it works here that “I decided”. And I'll tell you what happened.

It’s been a long time since I simply stopped being “nervous” during searches. Sorry if I pressed you. These are very new tools for me. It seems I just forgot to register the dependencies in order to load the boxes at all =)

**Reply by Gavolot** [reply]
You were right! But I just couldn’t understand “what exactly do you mean” because I didn’t add the dependency to “Cargo.toml”

**Comment by Gavolot** [comment]
Thank you all for your help, with perseverance I finally found what was going on.... This was a real test for me “considering that I am an experienced programmer,” but on all other tools, it was very difficult for me for the last 8 hours.
